src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx

```diff
@@ -1,545 +1,563 @@
+  const distribution = useAppSelector((state) => selectDistribution(state));
+  const customRepositories = useAppSelector((state) =>
+    selectCustomRepositories(state)
   );
-  const [selectedChosenPackages, setSelectedChosenPackages] = useState(
-    new Set()
+  const packages = useAppSelector((state) => selectPackages(state));
+
+  const [perPage, setPerPage] = useState(10);
+  const [page, setPage] = useState(1);
+  const [toggleSelected, setToggleSelected] = useState('toggle-available');
+
+  /*FOLLOW UP
+  const [toggleSourceRepos, setToggleSourceRepos] = useState(
+    'toggle-included-repos'
   );
-  const firstInputElement = useRef(null);
+  */
 
-  const oscapProfile = getState()?.values?.['oscap-profile'];
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [
+    searchRpms,
+    { data: dataCustomPackages, isSuccess: isSuccessCustomPackages },
+  ] = useSearchRpmMutation();
 
-  const { data: customizations, isSuccess: isSuccessCustomizations } =
-    useGetOscapCustomizationsQuery(
+  const { data: dataDistroPackages, isSuccess: isSuccessDistroPackages } =
+    useGetPackagesQuery(
       {
-        distribution: getState()?.values?.['release'],
-        profile: oscapProfile,
+        distribution: distribution,
+        architecture: arch,
+        search: searchTerm,
       },
-      {
-        skip: !oscapProfile,
-      }
+      { skip: !searchTerm }
     );
-  useEffect(() => {
-    if (customizations && customizations.packages && isSuccessCustomizations) {
-      const oscapPackages = {};
-      for (const pkg of customizations.packages) {
-        oscapPackages[pkg] = { name: pkg };
-      }
-      updateState(oscapPackages);
-    }
-  }, [customizations, isSuccessCustomizations, updateState]);
 
-  // this effect only triggers on mount
   useEffect(() => {
-    if (selectedPackages) {
-      const newChosenPackages = {};
-      for (const pkg of selectedPackages) {
-        newChosenPackages[pkg.name] = pkg;
-      }
-      setChosenPackages(newChosenPackages);
-    }
-  }, []);
-
-  useEffect(() => {
-    if (isSuccess) {
-      firstInputElement.current?.focus();
-    }
-  }, [isSuccess]);
-
-  const searchResultsComparator = useCallback((searchTerm) => {
-    return (a, b) => {
-      a = a.name.toLowerCase();
-      b = b.name.toLowerCase();
-
-      // check exact match first
-      if (a === searchTerm) {
-        return -1;
-      }
-
-      if (b === searchTerm) {
-        return 1;
-      }
-
-      // check for packages that start with the search term
-      if (a.startsWith(searchTerm) && !b.startsWith(searchTerm)) {
-        return -1;
-      }
-
-      if (b.startsWith(searchTerm) && !a.startsWith(searchTerm)) {
-        return 1;
-      }
+    const fetchCustomPackages = async () => {
+      await searchRpms({
+        apiSearchRpmRequest: {
+          search: searchTerm,
+          urls: customRepositories.flatMap((repo) =>
+            repo.baseurl ? repo.baseurl : ''
```

Comment:
This solves the Typescript error but I think that this adds an empty string to the array in the case that `repo.baseurl === undefined`, which I don't think we want.

But what worries me a bit more is that we will be failing silently. We can't properly search without a baseurl, can we? Which means that if a baseurl is missing from a repo there is a big problem. If we fail silently the user simply won't get the results that they are expecting and will be confused as to why. Instead, I think we should just fail loudly so the user knows there is a problem and can report a bug (since this should never happen). Does that sound reasonable?

```typescript
  const fetchCustomPackages = async () => {
    await searchRpms({
      apiSearchRpmRequest: {
        search: searchTerm,
        urls: customRepositories.flatMap((repo) => {
          if (!repo.baseurl) {
            throw new Error(
              `repo (id: ${repo.id}, name: ${repo?.name}) is missing baseurl`
            );
          }
          return repo.baseurl;
        }),
      },
    });
  };
```

Reply:
Great point, thank you! I've updated the code.